Replace isset with get

dshanske opened the pull request:
Ask and ye shall receive

dshanske committed:
Replace isset with get

snarfed commented:
ahhh this is so good! thank you!!! mind switching all of the existing `isset`s to it? not urgent.

i'm also tempted to move it out of the class, put i guess that puts it in the global PHP namespace, which is a bad idea...? yeah probably never mind. :P

snarfed commented:
friendly ping! this would be great to get in.

dshanske commented:
We can put it into the global namespace if we prefix it

snarfed commented:
eh, i'm fine with it as a static method, unless you feel strongly otherwise.

dshanske commented:
@snarfed I cleaned out a bunch of get related errors with undefined indexes and changed the signature of the get function to be more array index friendly. 

Also found the same variable name was used in two different contexts by accident, so changed it.

I did make one change that isn't strictly in the PR notification. I changed serialized in the error_log to json_encode because I found it easier to read. I think I originally put the serialize in.

snarfed commented on micropub.php:
```diff
@@ -925,28 +932,27 @@ public static function store_micropub_auth_response( $args ) {
-				$args['meta_input'] = array();
-			}
-			$type = static::$input['type'];
+		if ( ! isset( $args['ID'] ) && $props ) {
```
why was `! isset( $args['ID'] )` added here? seems unrelated?

dshanske replied:
Because before, it would run the remaining lines, which don't apply on creation, generating an error.

dshanske replied:
When I went to turn it into get, I realized that it was trying to run the replacement code on create.

snarfed replied:
you mean it was running the create code on replace? (and delete and add?) i think that was ok as is, but this is fine too. ok!

snarfed reviewed (changes requested):
hey, thanks! lgtm other than one errant change.
